File: Entrega2/NumericMethodsJava/Lagrange.java
/**
 *
 * @author Valeria
 */
public class Lagrange {

    public static void lagrange(double x[], double y[], int n) {
        double [] aux = new double[n];
        double den = 0;
        double mul = 1;
        for (int i = 0; i < n; i++) {
            System.out.print("L" + i + "= ");
            for (int j = 0; j < n; j++) {
                if(i!=j){
                    den = (x[i]-x[j]);
                    mul=mul*den;
                    System.out.print("(x - " + x[j] + ")");
                }
            }
            System.out.println(" " + ((1/mul)*y[i]) +"\n");
            mul=1;
        }
        
    }

    public static void main(String[] args) {
        double x[] = {-2,-1,2,3};
        double y[] = {12.13533528, 6.367879441, -4.610943901, 2.085536923};
        int n = x.length;

        lagrange(x, y, n);
    }
}

File: Entrega2/NumericMethodsPython/Lagrange.py


#Author Valeria

def lagrange(x,y,n):
    aux = []
    den = 0
    mul = 1
    for i in range(n):
        print("L" , i , "= ", end=" ")
        for j in range(n):
            if(i!=j):
                den = (x[i]-x[j])
                mul=mul*den
                print("(x - " , x[j] , ")", end=" ")
            
        
        print(" " , ((1/mul)*y[i]) ,"\n")
        mul=1
    


x = [-2,-1,2,3]
y = [12.13533528, 6.367879441, -4.610943901, 2.085536923]
n = len(x)

lagrange(x, y, n);
